test_suite/tests/test_macros.rs

```diff
@@ -864,3 +864,300 @@ fn test_packed_struct_can_derive_serialize() {
+            Token::I32(42),
+            Token::Str("field_two"),
+            Token::Str("hello"),
+            Token::StructEnd,
+        ],
+    );
+
+    // Test deserializing with UPPERCASE aliases
+    assert_de_tokens(
+        &TestStructUppercase {
+            field_one: 42,
+            field_two: "hello".to_string(),
+        },
+        &[
+            Token::Struct {
+                name: "TestStructUppercase",
+                len: 2,
+            },
+            Token::Str("FIELD_ONE"),
+            Token::I32(42),
+            Token::Str("FIELD_TWO"),
+            Token::Str("hello"),
+            Token::StructEnd,
+        ],
+    );
+}
+
+#[test]
+fn test_alias_all_camel_case() {
+    #[derive(Debug, PartialEq, Deserialize)]
+    #[serde(alias_all = "camelCase")]
+    struct TestStructCamelCase {
+        field_one: i32,
+        field_two: String,
+    }
+
+    // Test deserializing with original snake_case names
+    assert_de_tokens(
+        &TestStructCamelCase {
+            field_one: 42,
+            field_two: "hello".to_string(),
+        },
+        &[
+            Token::Struct {
+                name: "TestStructCamelCase",
+                len: 2,
+            },
+            Token::Str("field_one"),
+            Token::I32(42),
+            Token::Str("field_two"),
+            Token::Str("hello"),
+            Token::StructEnd,
+        ],
+    );
+
+    // Test deserializing with camelCase aliases
+    assert_de_tokens(
+        &TestStructCamelCase {
+            field_one: 42,
+            field_two: "hello".to_string(),
+        },
+        &[
+            Token::Struct {
+                name: "TestStructCamelCase",
+                len: 2,
+            },
+            Token::Str("fieldOne"),
+            Token::I32(42),
+            Token::Str("fieldTwo"),
+            Token::Str("hello"),
+            Token::StructEnd,
+        ],
+    );
+}
+
+#[test]
+fn test_alias_all_enum() {
+    #[derive(Debug, PartialEq, Deserialize)]
+    #[serde(alias_all = "kebab-case")]
+    enum TestEnum {
+        VariantOne { field_one: i32 },
+        VariantTwo { field_two: String },
+    }
+
+    // Test deserializing with original PascalCase names
+    assert_de_tokens(
+        &TestEnum::VariantOne { field_one: 42 },
+        &[
+            Token::Enum { name: "TestEnum" },
+            Token::Str("VariantOne"),
+            Token::Struct {
+                name: "VariantOne",
+                len: 1,
+            },
+            Token::Str("field_one"),
+            Token::I32(42),
+            Token::StructEnd,
+        ],
+    );
+
+    // Test deserializing with kebab-case aliases
+    assert_de_tokens(
+        &TestEnum::VariantOne { field_one: 42 },
+        &[
+            Token::Enum { name: "TestEnum" },
+            Token::Str("variant-one"),
+            Token::Struct {
+                name: "VariantOne",
+                len: 1,
+            },
+            Token::Str("field-one"),
```

Comment:
I would expect, that `alias_all` for enums will influence only variant names. If you need to appli the same rules to struct variants, you should add attribute to struct variant. If I'm not wrong, it is how `rename_all` attribute work.